File: 4th Semester/Design and Programming Environments/labs/4th lab/Java/BasketServices/java/IService.java
package ro.mpp2024;

import java.util.List;
import java.util.Optional;
import ro.mpp2024.IObserver;


public interface IService {
    User login(String username, String password, IObserver observer)throws Exception;
    void logout(int id, IObserver client);
    void signup(String name, String username, String password);
    List<Purchase> find_by_client(String client);
    List<Game> showcase_games_by_type(String type);
    void add_purchase(Purchase purchase) throws Exception;
    Game update_seats(Game game, int seats);
    void validatePassword(String password);
    List<Game> show_games();
    boolean find_by_username(String username);
}

File: 4th Semester/Design and Programming Environments/labs/4th lab/C#/BasketServices/IService.cs
namespace BasketServices
{
    using BasketModel;
    using System;
    using System.Collections.Generic;

    public interface IService
    {
        void Login(string username, string password, IObserver observer);
        void Logout(string username, IObserver client);
        void Signup(string name, string username, string password);
        List<Purchase> FindByClient(string client);
        List<Game> ShowcaseGamesByType(string type);
        void AddPurchase(Purchase purchase);
        Game UpdateSeats(Game game, int seats);
        void ValidatePassword(string password);
        List<Game> ShowGames();
        bool FindByUsername(string username);
    }
}
